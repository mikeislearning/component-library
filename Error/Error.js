// @format
import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Text, View } from 'react-native';

import { Button } from 'app/components';
import { resetState } from 'app/actions';
import store from 'app/stores';
import { sendSentryError } from 'app/services';

import styles from './Error.styles';

export default class Error extends Component {
  static propTypes = {
    error: PropTypes.oneOfType([PropTypes.string, PropTypes.object]),
  };

  static defaultProps = {
    error: 'An error occurred',
  };

  componentDidMount() {
    const { error } = this.props;
    sendSentryError('Looks like an error occurred', error);
  }

  reloadApp = () => {
    store.dispatch(resetState());
  };

  render() {
    return (
      <View style={styles.container}>
        <Text style={styles.text}>Looks like there was an error. Please reload the app.</Text>
        <Button onPress={this.reloadApp}>Reload App</Button>
      </View>
    );
  }
}
