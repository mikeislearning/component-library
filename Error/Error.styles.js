import { StyleSheet } from 'react-native';

import { colors, fonts } from 'app/theme';

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'center',
    marginVertical: 50,
    padding: 50,
  },
  text: {
    color: colors.arriveBlack,
    fontFamily: fonts.family.messinaSans.regular,
    fontSize: 16,
    marginBottom: 20,
    textAlign: 'center',
  },
});

export default styles;
