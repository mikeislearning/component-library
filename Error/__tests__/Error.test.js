/**
 * @format
 */

import React from 'react';
import { render } from 'react-native-testing-library';

import Error from '../Error';

describe('<Error />', () => {
  it('renders correctly', () => {
    const error = render(<Error error={'error'} />);
    const json = error.toJSON();
    expect(error).toMatchSnapshot();
    expect(json.type).toEqual('View');
    expect(json.children[0].type).toEqual('Text');
    expect(json.children[0].children[0]).toEqual('Looks like an error occurred :(');
  });
});
