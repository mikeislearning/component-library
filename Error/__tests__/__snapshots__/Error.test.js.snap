// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Error /> renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "flex": 1,
        "justifyContent": "center",
      },
      Object {
        "flexDirection": "row",
        "justifyContent": "space-around",
        "padding": 10,
      },
    ]
  }
>
  <Text
    style={
      Object {
        "color": "#CC2E41",
      }
    }
  >
    Looks like an error occurred :(
  </Text>
</View>
`;
